toolchain/check/handle_class.cpp

```diff
@@ -542,6 +542,13 @@ auto HandleParseNode(Context& context, Parse::BaseDeclId node_id) -> bool {
        .index = SemIR::ElementIndex(
            context.args_type_info_stack().PeekCurrentBlockContents().size())});
 
+  if (base_info.type_id != SemIR::TypeId::Error) {
+    auto base_class_info = context.classes().Get(
+        context.types().GetAs<SemIR::ClassType>(base_info.type_id).class_id);
+    CARBON_CHECK(!class_info.is_dynamic);
+    class_info.is_dynamic = base_class_info.is_dynamic;
```

Comment:
Sounds good, thanks!

In general: we (ideally) don't want it to be possible for input to trigger a CHECK fail, in part because our fuzzer will eventually find such cases. In cases where support for something is missing / broken, one tool we have available is to use `context.TODO` to report a diagnostic and then return `false` from the `Handle` function (which will cause us to bail out of the overall checking process).